src/backend/common/queries/dict_converters/insight_converter.py

```diff
@@ -0,0 +1,37 @@
+import json
+from typing import Dict, List, NewType
+
+from backend.common.consts.api_version import ApiMajorVersion
+from backend.common.models.insight import Insight
+from backend.common.queries.dict_converters.converter_base import ConverterBase
+
+InsightDict = NewType("InsightDict", Dict)
+
+
+class InsightConverter(ConverterBase):
+    SUBVERSIONS = {  # Increment every time a change to the dict is made
+        ApiMajorVersion.API_V3: 0,
+    }
+
+    @classmethod
+    def _convert_list(
+        cls, model_list: List[Insight], version: ApiMajorVersion
+    ) -> List[InsightDict]:
+        INSIGHT_CONVERTERS = {
+            ApiMajorVersion.API_V3: cls.insightsConverter_v3,
+        }
+        return INSIGHT_CONVERTERS[version](model_list)
+
+    @classmethod
+    def insightsConverter_v3(cls, insights: List[Insight]) -> List[InsightDict]:
+        return list(map(cls.insightConverter_v3, insights))
+
+    @classmethod
+    def insightConverter_v3(cls, insight: Insight) -> InsightDict:
+        return InsightDict(
+            {
+                "name": insight.name,
+                "data_json": json.loads(insight.data_json),
```

Comment:
should this just be `"data": insight.data`?

Also the contents of data isn't well enforced / consistent across insights. Not sure how that will affect the openapi definition in the future. It might be worth going through and strongly typing the insights data (doesn't have to be in this PR, but worth keeping in mind).

Reply:
I bet we could have one generic category of "top N" metrics, which are all basically the same shape - most banners, most matches played, etc are all the same effective shape. That schema would cover most of them. Then there are others that are more rare, like HOF or CMP winners, which I bet we could generify too.